apps/src/code-studio/headerShare.js

```diff
@@ -35,7 +35,8 @@ export function shareProject(shareUrl) {
 
     // Allow publishing for any project type that students can publish.
     // Younger students can now get to the share dialog if their teacher allows
-    // it, and should be able to publish in that case.
+    // it. ShareAllowedDialog will disable publishing, even if canPublish is true,
+    // if the project is in restricted share mode.
     const canPublish =
       !!appOptions.isSignedIn && AllPublishableProjectTypes.includes(appType);
```

Comment:
Naive question: should we maybe not set `canPublish` to `true` here if the project is in restricted share mode?  Why is it the dialog that does the prevention?

Reply:
I waffled on this. The reason I decided to do it this way is we already needed to get `inRestrictedShareMode` in `ShareAllowedDialog` to show an information message and hide the twitter/facebook buttons. I also thought the logic made more sense if we would never show the publish button if we are in restricted share mode, which meant we were already going to combine `inRestrictedShareMode` with `canPublish`. But I could check it here and check it again inside `ShareAllowedDialog` (in case we end up re-using `ShareAllowedDialog` somewhere else).